src/tests/multihost/alltests/test_krb_access_provider.py

```diff
@@ -0,0 +1,257 @@
+               "authentication success not found in /var/log/secure"
+        assert "Accepted password for foo3" in secure_log_str, \
+               "Accepted password not found in /var/log/secure"
+        assert f"Access allowed for user [foo3@{ds_instance_name}]" in sssd_log_str, \
+               "Access allowed not found in /var/log/sssd/sssd_example1.log"
+
+    @staticmethod
+    def test_0003_k5login_user3_with_user4(multihost):
+        """
+        :title: .k5login file has user principals for user foo3 and foo4.
+        :id: 3ddb4948-6b5a-4e82-85f7-c896bce14dda
+        :setup:
+          1. Clear the secure log, clear cache and restart sssd.
+          2. Add both the user principal in the k5login file
+        :steps:
+          1. Authenticate the user foo3 from the client
+          2. Check the /tmp/accessprovider.out file for expected messages.
+        :expectedresults:
+          1. User foo3 should be able to successfully login
+          2. /tmp/accessprovider.out file contains the expected output:
+             'foo3' username should be present in the file.
+        :teardown:
+          1. Clear the .k5login file.
+          2. Remove the /tmp/accessProvider.out file
+        """
+        multihost.client[0].run_command("truncate -s 0 /var/log/secure", raiseonerr=False)
+        client = sssdTools(multihost.client[0])
+        client.clear_sssd_cache()
+
+        echo_cmds = ['echo "foo3@EXAMPLE.TEST" > /home/foo3/.k5login',
+                     'echo "foo4@EXAMPLE.TEST" >> /home/foo3/.k5login',
+                     'restorecon -v /home/foo3/.k5login']
+
+        for command in echo_cmds:
+            cmd = multihost.client[0].run_command(command, raiseonerr=False)
+
+        user = 'foo3'
+        ssh_output = client.auth_from_client(user, 'Secret123')
+
+        client_hostname = multihost.client[0].sys_hostname
+        ssh = pexpect_ssh(client_hostname, 'foo4', 'Secret123', debug=False)
+        ssh.login(login_timeout=10, sync_multiplier=1, auto_prompt_reset=False)
+        ssh.command(f"ssh -o StrictHostKeyChecking=no -o PasswordAuthentication=no foo3@{client_hostname}\
+                     'id > /tmp/accessProvider_id_krb5_003.out 2>&1' ")
+        ssh.logout()
+
+        logfile = multihost.client[0].get_file_contents("/tmp/accessProvider_id_krb5_003.out").decode('utf-8')
+
+        multihost.client[0].run_command("truncate -s 0 /home/foo3/.k5login", raiseonerr=False)
+        multihost.client[0].run_command("rm -rf /tmp/accessProvider_id_krb5_003.out", raiseonerr=False)
+        assert cmd.returncode == 0, f"{command} did not execute successfully"
+        assert ssh_output == 3, "foo3 failed to log in"
+        assert "foo3" in logfile, "foo3 not found in /tmp/accessProvider_id_krb5_003.out"
+
+    @staticmethod
+    def test_0004_k5login_user4_with_deleted_user3(multihost):
+        """
+        :title: .k5login has principals for foo3 and foo4 but user foo3 is deleted
```

Comment:
IDM-SSSD-TC: krb_provider: krb_access_provider: k5login has testuser3 and testuser4 and testuser3 is deleted